pkg/river/vm/vm.go

```diff
@@ -131,6 +132,9 @@ func (vm *Evaluator) evaluateBlockOrBody(scope *Scope, assoc map[value.Value]ast
 		Assoc:   assoc,
 		TagInfo: ti,
 	}
+	//TODO(ptodev): if Decode() returns an error which is not a diagnostic, we should assume that
+	// it was because the error applies to the whole block and we should repackage this error here
+	// as a diagnostic with the start and end position of the block.
```

Comment:
@rfratto We had a discussion on Slack where I mentioned that `sd.Decode` doesn't have the brace position of the block because we only pass the body and not the block itself. You suggested to pass the block to `sd.Decode` insteadn of the body. But in the case of evaluating the main body of the river file, we would not have a root level block and root level braces. Hence we cannot **only** pass a block to the `sd.Decode` function. 

I am not sure what is the best way to get `sd.Decode` to print diagnostic info with curly brace start and end position of the block for the `missing required attribute` and `missing required block` errors.

The easiest and most hacky solution feels like accepting an `ast.Node` as a parameter for `sd.Decode`, but I wonder if there is a more type safe way to do it which doesn't involve a type assertion.

Also, what if we have a `missing required attribute` or `missing required block` error on the root level, where there is no block? I guess we should print the start and end position of the whole river config?

Personally, I think the easiest solution is to report both the start and end position in the error message, and keep using the body instead of the block at all times for `sd.Decode`. Then, even though we don't report the position of the curly braces it'd be clear to the user that we are erroring on the whole range of the block. However, I think you had reservations for making the non-pretty error printing include more information?